schema.sql

```diff
@@ -13,258 +18,301 @@ drop table if exists areas;
-	balancing_topology_from text references balancing_topologies(name),
-	balancing_topology_to text references balancing_topologies(name),
-	area_from text references areas(name),
-	area_to text references areas(name),
-	continuous_rating float not null check(continuous_rating >= 0),
-	ste_rating float not null check (ste_rating >=0),
-	lte_rating float not null check (lte_rating >=0),
-	line_length float not null check (line_length >= 0)
+-- physical connection between entities
+create table arcs(
+        id integer primary key
+        ,from_to int
+        ,to_from int
+        ,foreign key (from_to) references entities(id)
+        ,foreign key (to_from) references entities(id)
 );
 
+-- transmission lines
+create table transmission_lines (
+        id integer primary key
+        ,arc_id int
+	,continuous_rating real not null check(continuous_rating >= 0)
+	,ste_rating real not null check (ste_rating >=0)
+	,lte_rating real not null check (lte_rating >=0)
+	,line_length real not null check (line_length >= 0)
+        ,foreign key (arc_id) references arcs(id)
+) strict;
+
 
--- ) strict;
 -- flow between two regions
-create table  transmission_interchange (
-	area_from text not null references areas(name),
-	area_to text not null references areas(name),
-	max_flow_from float not null,
-	max_flow_to float not null
-);
+create table transmission_interchange (
+        arc_id int references arcs(id)
+        ,max_flow_from real not null
+	,max_flow_to real not null
+) strict;
 
--- create load input table, at some point need to add the growth rate
-create table  demand_requirements (
-	entity_attribute_id integer primary key,
-	peak_load float not null,
-	area text references areas(name),
-	balancing_topology text references balancing_topologies(name)
+-- create supplemental atttributes
+create table supplemental_attributes (
+	id integer primary key
+        ,type text not null
+	,value jsonb not null
 );
 
--- create entity-attribute table
-create table attributes (
-	entity_attribute_id integer primary key,
-	entity_id integer not null,
-	entity_type text not null,
-	name text not null,
-	value any null,
-	data_type text not null references data_types(name),
-	foreign key (entity_type) references table_names(name),
-    check (name != 'entity_id')
+-- association table between attributes and entities
+create table attributes_associations(
+	attribute_id integer not null
+	,entity_id integer not null
+	,foreign key (entity_id) references entities(id)
+	,foreign key (attribute_id) references supplemental_attributes(id)
+) strict;
+
+create table time_series(
+        id integer primary key
+        ,uuid text null
+        ,time_series_type text not null
+        ,name text not null
+        ,initial_timestamp datetime not NULL
+        ,resolution_ms integer not NULL
+        ,horizon integer not null
+        ,interval integer not null
+        ,length integer not null
+        ,features jsonb null
+        ,metadata jsonb null
```

Comment:
So time-series are not a type of attributes? Why?

Reply:
What do you mean type attributes?